### PR TITLE
fix: use int instead of Token to decouple extractor and features

### LINKED ISSUE #1139: dnlib dependancy in capa/features/address.py
### Context
First of all, I'm a big fan of Capa, thank your for this great tool! I have been successfully integrating Capa into [Malcat](https://malcat.fr) (you can test the capa integration in the free version if you're curious) for some time now. I use my own main script to list matching rules and my own extractor (using Malcat's python bindings). So basically I just keep the engine and the rules and discard evtg else. Integration was pretty easy and clean until now because the engine/features are very well detached from the extractors (great job guys !).

### Problem

In capa 4.0, the separation between extractor and features has a tiny flaw: there is a dnlib dependancy in `capa/features/address.py`:

```python
[capa/features/address.py](https://github.com/mandiant/capa/compare/v3.2.0...master#diff-0727ca1e119ebd353a0204473ea5b404b0e0d60d58d777d6d81af2ad75af8732)
@@ -0,0 +1,117 @@
import abc

from dncil.clr.token import Token    # <-- there
```

And this is used in these two classes:

```python
class DNTokenAddress(Address):
    """a .NET token"""

    def __init__(self, token: Token):
        self.token = token
    # ...


class DNTokenOffsetAddress(Address):
    """an offset into an object specified by a .NET token"""

    def __init__(self, token: Token, offset: int):
        assert offset >= 0
        self.token = token
        self.offset = offset
    # ...
```

The use of dnlib is specific to the current dnlib-based extractor, another extractor could use another library. In fact, I have my own dotnet parser and my extractor doesn't need dnlib at all. And that's unfortunate since only the `token.value` seems to be used there, so why not store directly the token value as `int`?  In fact, I made the change in the v4.0 branch and it seems to have no impact at all.

Of course I could create a parallel branch with the `dncil.clr.token.Token`->`int` change and always rebase when integrating capa (that's what I'm doing right now), but I find it unfortunate that the clean features/extractors separation is gone because of this small detail. 

If you want, I could submit a PR where `int` is used there instead of `dncil.clr.token.Token`.

### PR DESCRIPTION
**Pull request by mr-tz**
closes #1139

### Checklist

- [ ] No CHANGELOG update needed

- [ ] No new tests needed

- [ ] No documentation update needed

### CODE REVIEW

File: capa/features/address.py [hunk 1]
```diff
@@ -56,48 +54,48 @@ def __repr__(self):
 class DNTokenAddress(Address):
     """a .NET token"""
 
-    def __init__(self, token: Token):
+    def __init__(self, token: int):
         self.token = token
```
Comment (mr-tz) [inline capa/features/address.py:58]: I think we can update these like the other classes, adding:
`return int.__new__(cls, token)`

Comment (mr-tz) [reply capa/features/address.py]: c5b4fc33

Comment (mr-tz) [reply capa/features/address.py]: c5b4fc33

Comment (mr-tz) [reply capa/features/address.py]: updated

**Review by mike-hunhoff — APPROVED** [review]
really like how the decoupling cleaned up the code, thank you! LGTM 🚀

**Pull request merged**